stripe/src/main/java/com/stripe/android/model/PaymentIntent.java

```diff
@@ -0,0 +1,299 @@
+            shippingMethod = ShippingMethod.fromString(shippingString);
+        }
+        Map<String, Object> nextSourceAction = optMap(jsonObject, FIELD_NEXT_SOURCE_ACTION);
+        String source = optString(jsonObject, FIELD_SOURCE);
+
+        return new PaymentIntent(
+                id,
+                objectType,
+                allowedSourceTypes,
+                amount,
+                canceledAt,
+                captureMethod,
+                clientSecret,
+                confirmationMethod,
+                created,
+                currency,
+                description,
+                livemode,
+                nextSourceAction,
+                receiptEmail,
+                shippingMethod,
+                source,
+                status);
+    }
+
+    @NonNull
+    @Override
+    public JSONObject toJson() {
+        JSONObject jsonObject = new JSONObject();
+        putStringIfNotNull(jsonObject, FIELD_ID, mId);
+        putStringIfNotNull(jsonObject, FIELD_OBJECT, mObjectType);
+        putArrayIfNotNull(jsonObject, FIELD_ALLOWED_SOURCE_TYPES, listToJsonArray(mAllowedSourceTypes));
+        putLongIfNotNull(jsonObject, FIELD_AMOUNT, mAmount);
+        putLongIfNotNull(jsonObject, FIELD_CANCELED, mCanceledAt);
+        putStringIfNotNull(jsonObject, FIELD_CAPTURE_METHOD, mCaptureMethod);
+        putStringIfNotNull(jsonObject, FIELD_CLIENT_SECRET, mClientSecret);
+        putStringIfNotNull(jsonObject, FIELD_CONFIRMATION_METHOD, mConfirmationMethod);
+        putLongIfNotNull(jsonObject, FIELD_CREATED, mCreated);
+        putStringIfNotNull(jsonObject, FIELD_CURRENCY, mCurrency);
+        putStringIfNotNull(jsonObject, FIELD_DESCRIPTION, mDescription);
+        putBooleanIfNotNull(jsonObject, FIELD_LIVEMODE, mLiveMode);
+        putMapIfNotNull(jsonObject, FIELD_NEXT_SOURCE_ACTION, mNextSourceAction);
+        putStringIfNotNull(jsonObject, FIELD_RECEIPT_EMAIL, mReceiptEmail);
+        putStripeJsonModelIfNotNull(jsonObject, FIELD_SHIPPING, mShippingMethod);
+        putStringIfNotNull(jsonObject, FIELD_SOURCE, mSource);
+        putStringIfNotNull(jsonObject, FIELD_STATUS, mStatus);
+        return jsonObject;
+    }
+
+    @NonNull
+    @Override
+    public Map<String, Object> toMap() {
+        Map<String, Object> map = new HashMap<>();
+        map.put(FIELD_ID, mId);
+        map.put(FIELD_OBJECT, mObjectType);
+        map.put(FIELD_ALLOWED_SOURCE_TYPES, mAllowedSourceTypes);
+        map.put(FIELD_AMOUNT, mAmount);
+        map.put(FIELD_CANCELED, mCanceledAt);
+        map.put(FIELD_CLIENT_SECRET, mClientSecret);
+        map.put(FIELD_CAPTURE_METHOD, mCaptureMethod);
+        map.put(FIELD_CONFIRMATION_METHOD, mConfirmationMethod);
+        map.put(FIELD_CREATED, mCreated);
+        map.put(FIELD_CURRENCY, mCurrency);
+        map.put(FIELD_LIVEMODE, mLiveMode);
+        map.put(FIELD_NEXT_SOURCE_ACTION, mNextSourceAction);
+        map.put(FIELD_DESCRIPTION, mDescription);
```

Comment:
super nit: should this move up between currency & livemode to keep things in alphabetical order? Everywhere else you've done a great job, I briefly though description was missing because it wasn't where I expected